src/transformers/models/layoutlmv2/modeling_layoutlmv2.py

```diff
@@ -0,0 +1,1388 @@
+        elif isinstance(module, nn.Embedding):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.padding_idx is not None:
+                module.weight.data[module.padding_idx].zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+
+
+def my_convert_sync_batchnorm(module, process_group=None):
+    # same as `nn.modules.SyncBatchNorm.convert_sync_batchnorm` but allowing converting from `detectron2.layers.FrozenBatchNorm2d`
+    if isinstance(module, torch.nn.modules.batchnorm._BatchNorm):
+        return nn.modules.SyncBatchNorm.convert_sync_batchnorm(module, process_group)
+    module_output = module
+    if isinstance(module, detectron2.layers.FrozenBatchNorm2d):
+        module_output = torch.nn.SyncBatchNorm(
+            num_features=module.num_features,
+            eps=module.eps,
+            affine=True,
+            track_running_stats=True,
+            process_group=process_group,
+        )
+        module_output.weight = torch.nn.Parameter(module.weight)
+        module_output.bias = torch.nn.Parameter(module.bias)
+        module_output.running_mean = module.running_mean
+        module_output.running_var = module.running_var
+        module_output.num_batches_tracked = torch.tensor(0, dtype=torch.long, device=module.running_mean.device)
+    for name, child in module.named_children():
+        module_output.add_module(name, my_convert_sync_batchnorm(child, process_group))
+    del module
+    return module_output
+
+
+class VisualBackbone(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.cfg = detectron2.config.get_cfg()
+        add_layoutlmv2_config(self.cfg)
+        meta_arch = self.cfg.MODEL.META_ARCHITECTURE
+        model = META_ARCH_REGISTRY.get(meta_arch)(self.cfg)
+        assert isinstance(model.backbone, detectron2.modeling.backbone.FPN)
+        self.backbone = model.backbone
+        if (
+            config.convert_sync_batchnorm
+            and torch.distributed.is_available()
+            and torch.distributed.is_initialized()
+            and torch.distributed.get_rank() > -1
+        ):
+            self_rank = torch.distributed.get_rank()
+            node_size = torch.cuda.device_count()
+            world_size = torch.distributed.get_world_size()
+            assert world_size % node_size == 0
+
+            node_global_ranks = [
+                list(range(i * node_size, (i + 1) * node_size)) for i in range(world_size // node_size)
+            ]
+            sync_bn_groups = [
+                torch.distributed.new_group(ranks=node_global_ranks[i]) for i in range(world_size // node_size)
+            ]
+            node_rank = self_rank // node_size
+            assert self_rank in node_global_ranks[node_rank]
+
+            self.backbone = my_convert_sync_batchnorm(self.backbone, process_group=sync_bn_groups[node_rank])
```

Comment:
This code has to do with distributed training, probably to be removed.

Reply:
Can be put in a method otherwise, so a user can activate it when they are doing distributed training.